server/matchmaker/algorithm/matchmaker.py

```diff
@@ -0,0 +1,43 @@
+from typing import Dict, Iterable, List, Set, Tuple
+
+from ...decorators import with_logger
+from ..search import Match, Search
+
+
+@with_logger
+class Matchmaker(object):
+    def find(
+        self, searches: Iterable[Search], team_size: int
+    ) -> Tuple[List[Match], List[Search]]:
+        raise NotImplementedError(
+            "Matchmaker.find should be implemented by concrete subclasses"
+        )
```

Comment:
You can use the `abc.ABC` (Abstract Base Class) base class to raise these exceptions for you